src/gsy_e/models/strategy/__init__.py

```diff
@@ -774,9 +774,10 @@ def update_bid_rates(self, market: "TwoSidedMarket", updated_rate: float) -> Non
         """Replace the rate of all bids in the market slot with the given updated rate."""
         existing_bids = list(self.get_posted_bids(market))
         for bid in existing_bids:
+            if abs(bid.energy_rate - updated_rate) <= FLOATING_POINT_TOLERANCE:
+                continue
             assert bid.buyer == self.owner.name
-            if bid.id in market.bids.keys():
-                bid = market.bids[bid.id]
```

Comment:
I did not see the need for this additional lookup so I removed it and the tests pass. However, please let me know if you think this is important and explain why. Thanks!

Reply:
One could even go further and use `market.delete_bid(bid)` instead of `market.delete_bid(bid.id)`. Outside of the scope of the change though therefore leave it as is.